internal-enrichment/import-external-reference/src/import-external-reference.py

```diff
@@ -45,18 +45,92 @@ def __init__(self):
             True,
         )
         self.headers = {"User-Agent": "Mozilla/5.0 (Windows NT 6.1; Win64; x64)"}
-        self.wkhtmltopdf_path = get_config_variable(
-            "WKHTMLTOPDF_PATH",
-            ["import_external_reference", "wkhtmltopdf_path"],
-            config,
-        )
 
     def delete_files(self):
         if os.path.exists("data.html"):
             os.remove("data.html")
         if os.path.exists("data.pdf"):
             os.remove("data.pdf")
 
+    @staticmethod
+    def _is_cookies_accepted(page):
+        """
+        Attempts to accept cookies on a web page.
+        This method searches for various elements (links and buttons) that are commonly used for cookie consent prompts.
+        It uses a list of predefined selectors to identify and click on elements that contain text variations related to cookie acceptance.
+
+        :param page: The Playwright Page object representing the current web page.
+        :return: bool: Returns True if a cookie acceptance element was successfully clicked, otherwise returns False.
+        """
+
+        found = False
+
+        selectors = [
+            'a:has-text("agree")',
+            'button:has-text("agree")',
+            'a:has-text("accept all")',
+            'button:has-text("accept all")',
+            'a:has-text("accept")',
+            'button:has-text("accept")',
+            'a:has-text("i agree")',
+            'button:has-text("i agree")',
+            'a:has-text("allow")',
+            'button:has-text("allow")',
+        ]
+
+        for selector in selectors:
+            try:
+                page.locator(selector).first.click(timeout=1000, force=True)
+                found = True
+                break
+            except:
+                continue
```

Comment:
I think you could only except playwright error here and log message with debug level.

https://playwright.dev/python/docs/api/class-error [consulted on October 7th, 2024]